browser/css/jssidebar.css

```diff
@@ -423,3 +431,30 @@ span.jsdialog.sidebar.ui-treeview-notexpandable {
 #TemplatePanel > div:nth-child(4) {
 	flex-grow: 1;
 }
+
+#TemplatePanel > div:first-child {
+	order: 1;
+}
+
+#TemplatePanel #settings {
+	order: 2;
+}
+
+#TemplatePanel #filter {
```

Comment:
Plese do this change in the core (.ui file which describes the LO UI, there is graphical tool for that Glade).
So here in online please remove "reordering" part.

file is: sfx2/uiconfig/ui/templatepanel.ui  (my last commit there: https://gerrit.libreoffice.org/c/core/+/179625/2/sfx2/uiconfig/ui/templatepanel.ui)

you need core dev setup (LO)